### DISCUSSION TITLE
Shorthand function support

### DISCUSSION THREAD

**Discussion by blahbleepew** [adelsz/pgtyped#489] [General]
Is it possible to support a shorthand form? The way the functions are generated it takes a lot of linespace, especially when you have code formatters which want to put each argument on a separate line:

```javascript
const user = await userData.run(
  {
    userId,
  },
  client
);
```

Ideally it would be nice to have something like this (if possible to get rid of .run as well):

```javascript
const user = await userData(client, {
  userId,
});
```

**Comment by adelsz** [comment]
Making changes to the query signature is not planned right now. Instead the plan is to integrate the `pg.Client` runtime into pgTyped queries. Moved this issue to discussions to collect feedback on what the final API for such an integration should be.

The API I have in mind will look something like this:
```ts
import {getUserData} from './queries';
// ...

const userData = await getUserData({ userId: 1 });
```

Internally `getUserData` will be using a singleton global `pg.Client` or `pg.Pool` instance.

There are a couple of important questions:

1. Where to define pg Client and Pool configuration?
2. How to use transactions with the new syntax?
3. What should the API for accessing the underlying client look like?

**Comment by pokedotdev** [comment]
I solved this problem by creating a wrapper
```ts
import type { PreparedQuery } from '@pgtyped/runtime'

const query = <P, R>(query: PreparedQuery<P, R>, params: P) => {
	return query.run(params, client)
}

const user = await query(getUserData, {
	userId
})
```
